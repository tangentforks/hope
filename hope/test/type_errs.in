ord 23;

1 = 'a';

[1, 'a', 'b', 'c'];

dec f: alpha -> alpha;
--- f 1 <= 1;
