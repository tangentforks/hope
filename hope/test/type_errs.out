  ord 23
  ord : char -> num
  23 : num
line 1: type error - argument has wrong type
  1 = 'a'
  (=) : alpha # alpha -> bool
  1 : num
  'a' : char
line 3: type error - argument has wrong type
  [1, 'a', 'b', 'c']
  (::) : alpha # list alpha -> list alpha
  1 : num
  "abc" : list char
line 5: type error - argument has wrong type
  declared type: alpha -> alpha
  inferred type: num -> num
line 8: type error - 'f': does not match declaration
